### PR TITLE
Don't pass MapperService to LuceneReferenceResolver

### PR DESCRIPTION
**Pull request by romseygeek**
LuceneReferenceResolver double-checks that a passed in Reference points
to a concrete lucene field by looking at the mappings reported by an IndexShard's
MapperService.  However, references that point to non-ignored non-mapped
fields will already have been resolved as VoidReferences earlier in the call stack,
so we can replace this with an instanceof check and stop passing MapperServices
around.

### CODE REVIEW

File: server/src/main/java/io/crate/expression/reference/doc/lucene/LuceneReferenceResolver.java [hunk 1]
```diff
@@ -218,6 +187,19 @@ public LiteralValueExpression(Object value) {
         public Object value() {
             return value;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
```
Comment (matriv) [inline server/src/main/java/io/crate/expression/reference/doc/lucene/LuceneReferenceResolver.java:193]: Why did you add equals and hashCode, is it needed now?

**Review by matriv — APPROVED** [review]
LGTM, left a question.

File: server/src/main/java/io/crate/expression/reference/doc/lucene/LuceneReferenceResolver.java (hunk 1 above)
Comment (romseygeek) [reply server/src/main/java/io/crate/expression/reference/doc/lucene/LuceneReferenceResolver.java:193]: It made writing the test easier, but I can remove it and rework the test if you think that's better?

Comment (matriv) [reply server/src/main/java/io/crate/expression/reference/doc/lucene/LuceneReferenceResolver.java:193]: I just didn't see equals and hashcode on other classes, so If it's only for the tests I'd say either change the test or add a comment here to explain where they are used.

**Comment by romseygeek** [comment]
@Mergifyio squash title+body
